unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AudioSource/ECSAudioSourceComponentHandler.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Helpers;
+using DCL.Models;
+using DCL.SettingsCommon;
+using UnityEngine;
+using AudioSettings = DCL.SettingsCommon.AudioSettings;
+
+namespace DCL.ECSComponents
+{
+    public class ECSAudioSourceComponentHandler : IECSComponentHandler<ECSAudioSource>, IOutOfSceneBoundariesHandler
+    {
+        internal AudioSource audioSource;
+        internal AssetPromise_AudioClip promiseAudioClip;
+        
+        private long playedAtTimestamp = 0;
+        private bool isOutOfBoundaries = false;
+        
+        private ECSAudioSource model;
+        private IParcelScene scene;
+        private AudioClip audioClip;
+
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
+        {
+            audioSource = entity.gameObject.AddComponent<AudioSource>();
+
+            if (Settings.i != null)
```

Comment:
This is not an easy task and would require a major refactor to the new ECS system since these functions are common among the all the components but not all components needs those dependencies.

Reply:
Settings must be ported into a DataStore approach. All the data should live in the DataStore so a change there will have effect in the whole application.

The settings UI should just be a visual representation of that datastore.

Until then I agree with Adri, there's no easy way to circunvent this. Maybe you can port the AudioSettings into a dataStore approach? there are already some settings doing it, like the max avatars.

Reply:
That way we would be boy-scoutting the issue and making it smaller on each instance we face.

Reply:
I will try something